File: app/src/main/java/com/example/richtext/span/CustomLinkSpan.java
package com.example.richtext.span;


import android.content.Context;
import android.graphics.Typeface;
import android.text.TextPaint;
import android.text.TextUtils;
import android.text.style.ClickableSpan;
import android.view.View;

import com.shuyu.textutillib.listener.SpanUrlCallBack;
import com.shuyu.textutillib.span.LinkSpan;

/**
 * 自定义显示可点击的手机号码和URL
 * Created by shuyu on 2016/11/10.
 */
public class CustomLinkSpan extends LinkSpan {

    public CustomLinkSpan(Context context, String url, int color, SpanUrlCallBack spanUrlCallBack) {
        super(context, url, color, spanUrlCallBack);
    }

    @Override
    public void updateDrawState(TextPaint ds) {
        super.updateDrawState(ds);
        ds.setUnderlineText(true);
        ds.setTypeface(Typeface.DEFAULT_BOLD);
    }
}

File: app-kotlin/src/main/java/com/example/shuyu/span/CustomLinkSpan.kt
package com.example.shuyu.span


import android.content.Context
import android.graphics.Typeface
import android.text.TextPaint

import com.shuyu.textutillib.listener.SpanUrlCallBack
import com.shuyu.textutillib.span.LinkSpan

/**
 * 自定义显示可点击的手机号码和URL
 * Created by shuyu on 2016/11/10.
 */
open class CustomLinkSpan(context: Context, url: String, color: Int, spanUrlCallBack: SpanUrlCallBack) : LinkSpan(context, url, color, spanUrlCallBack) {

    override fun updateDrawState(ds: TextPaint) {
        super.updateDrawState(ds)
        ds.isUnderlineText = true
        ds.typeface = Typeface.DEFAULT_BOLD
    }
}